src/hdmf_zarr/backend.py

```diff
@@ -726,55 +726,16 @@ def __get_path(self, builder):
             path = "%s%s" % (delim, delim.join(reversed(names)))
         return path
 
-    @staticmethod
```

Comment:
This method was convoluted and not really used internally. I have simplified the code and its usage. This introduces a breaking change, but since we have not released 1.0 and I doubt users are using this method, I think it is OK to remove.